### PR TITLE
Get thumbnail as base64Image

### PR DESCRIPTION
**Pull request by vemarav**
#### frameQuality
Setting this prop will fire [onFrameChange](#onframechange) callback

* **0 (default)** - Don't display the video in fullscreen
* **1** - Display the video in fullscreen

Platforms: iOS, Android

---

#### onFrameChange
callback function that is called for each frame when [frameQuality](#framequality) > 0`, results each frame as base64Image(thumbnail)

```javascript
  <Video
    {...videoProps}
    frameQuality={0.3} // required to fire callback
    onFrameChange={(thumbnail: string) => setThumbnail(thumbnail)}
  />
```

Platform: iOS, Android

### CODE REVIEW

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 1]
```diff
@@ -244,12 +248,31 @@ public void handleMessage(Message msg) {
...
                     break;
             }
         }
     };
 
+    private void setFrame() {
+        if(frameQuality == null) return;
+        if(frameQuality > 1) frameQuality = 1;
+        frameQuality = Math.round(frameQuality * 100);
+        ((Runnable) () -> {
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:262]: Are you sure Runnable.run() will really delay the call ?
I think you should use a handler to do that. else it will just run here

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 2]
```diff
@@ -244,12 +248,31 @@ public void handleMessage(Message msg) {
                         }
                         msg = obtainMessage(SHOW_PROGRESS);
                         sendMessageDelayed(msg, Math.round(mProgressUpdateInterval));
+                        setFrame();
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:251]: Is it possible to decorrelate frame management & time position management. ?
If use set 10 msec for progress update because he wants to be accurate, I don't think he needs a new frame every 10ms.
You just need to accept an object in PROP_FRAME_QUALITY to add quality & frequency

File: ios/Video/RCTVideo.swift [hunk 3]
```diff
@@ -303,14 +335,17 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
...
+                    }
                     self._playerObserver.playerItem = self._playerItem
                     self.setPreferredForwardBufferDuration(self._preferredForwardBufferDuration)
                     self.setPlaybackRange(playerItem, withVideoStart: self._source?.startTime, withVideoEnd: self._source?.endTime)
                     self.setFilter(self._filterName)
                     if let maxBitRate = self._maxBitRate {
                         self._playerItem?.preferredPeakBitRate = Double(maxBitRate)
                     }
-
+
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift:348]: minor unnecessary spaces

File: ios/Video/RCTVideo.swift [hunk 4]
```diff
@@ -1130,8 +1171,8 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
                 }
             }
 
-            self.reactViewController().view.frame = UIScreen.main.bounds
-            self.reactViewController().view.setNeedsLayout()
+                self.reactViewController().view.frame = UIScreen.main.bounds
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift:1174]: unnecessary changes

File: ios/Video/RCTVideo.swift [hunk 5]
```diff
@@ -469,7 +504,7 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
 
     @objc
     func setSeek(_ info:NSDictionary!) {
-        let seekTime:NSNumber! = info["time"] as! NSNumber
+        let seekTime:NSNumber! = (info["time"] as! NSNumber)
```
Comment (freeboub) [inline ios/Video/RCTVideo.swift:507]: unnecessary changes ?

**Review by freeboub — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java (hunk 1 above)
Comment (vemarav) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:262]: Runnable won't delay but It doesn't block main thread or ui thread

**Comment by KrzysztofMoch** [comment]
Closed due to inactivity
